Update Pull Request Template [ci-skip] (#45495)
rails/rails: .github/pull_request_template.md

## About this template

This template aims to help contributors to write better descriptions.

> Feel free to discard it if you need to.

Reviewer: I think this really needs to be in a comment block too... if not, it will get left untouched often, and every PR notification will end up starting with the same words.

> Feel free to discard it if you need to.

Author: @eileencodes @matthewd I put them into a comment. I also added some more comments for motivation section.

### Motivation / Background

This Pull Request has been created because [REPLACE ME]

### Detail

This Pull Request changes [REPLACE ME]

### Additional information

<!-- Provide additional information such as benchmarks, reference to other repositories or alternative solutions. -->

### Checklist

Before submitting the PR make sure the following are checked:

> * [ ] This Pull Request is related to one thing.

Reviewer 2: ```suggestion
* [ ] This Pull Request is related to one change. Changes that are unrelated should be opened in separate PRs.
```

* [ ] There is no typo in commit messages and comments.
* [ ] Commit message contains `[Fix #issue-number]` (if the related issue exists).
* [ ] Feature branch is up-to-date with `main` (if not - rebase it).
* [ ] Related commits are squashed together.
* [ ] Tests are added if you fix a bug or add a feature.

> * [ ] CHANGELOG files are updated if requested.

Reviewer 2: ```suggestion
* [ ] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.
* [ ] PR is not in a draft state.
* [ ] CI is passing.
```

<!--
About this template

The following template aims to help contributors write a good description for their pull requests.
We'd like you to provide a description of the changes in your pull request (i.e. bugs fixed or features added), motivation behind the changes, and complete the checklist below before opening a pull request.

> Feel free to discard it if you need to (e.g. when you just fix a typo). -->

Reviewer 2: ```suggestion

Feel free to discard it if you need to (e.g. when you just fix a typo). -->
```
